python/ray/tests/spark/test_basic.py

```diff
@@ -80,6 +99,19 @@ def test_cpu_allocation(self):
                 self.max_spark_tasks // 2 + 1,
             ),  # Test case: requesting resources exceeding all cluster resources
         ]:
+            num_ray_task_slots = self.max_spark_tasks // (
+                num_cpus_worker_node // self.num_cpus_per_spark_task
+            )
+            (
+                mem_per_worker,
+                object_store_mem_per_worker,
+                _,
+            ) = _calc_mem_per_ray_worker_node(
+                num_task_slots=num_ray_task_slots,
+                physical_mem_bytes=_RAY_ON_SPARK_WORKER_PHYSICAL_MEMORY_BYTES,
+                shared_mem_bytes=_RAY_ON_SPARK_WORKER_SHARED_MEMORY_BYTES,
+                configured_object_store_bytes=None,
+            )
```

Comment:
I think this is not related to autoscaling? Given the PR is huge, let's remove non-autoscaling related changes.

Reply:
This is refinement for existing non-autoscaling mode tests, adding assertion for ray worker node memory allocation.

The reason is, in autoscaling PR, I made some refactor on existing non-autoscaling mode code, to ensure it does not break existing features, I want to make existing test more strict, so I added memory allocation assertion for non-allocation mode code.

It is just a small piece update in testing code, hope it does not increase reviewer burden.